### PR TITLE
Feature/fix fw issue

### PR DESCRIPTION
**Pull request by yuhei159**
This branch fixed issue described in #26.

### CODE REVIEW

File: library/niftycloud_fw.py [hunk 1]
```diff
@@ -683,6 +693,7 @@ def main():
 			log_filters       = dict(required=False, type='dict', default=dict()),
 			ip_permissions    = dict(required=False, type='list', default=list()),
 			state             = dict(required=False, type='str',  default='present', choices=['present']),
+			purge_rules       = dict(required=False, type='bool', default=True),
```
Comment (heriet) [inline library/niftycloud_fw.py]: Why new paramrter named `purge_rules` ?

In ec2_group, parameter is named purge_rules for rules and purge_rules_egress for rules_egress.

see: http://docs.ansible.com/ansible/ec2_group_module.html

File: library/niftycloud_fw.py [hunk 2]
```diff
@@ -587,6 +592,11 @@ def revoke_security_group(module, result, security_group_info):
 	if revoke_rules_size == 0:
 		return (result, security_group_info)
 
+	goal_rules_size    = len(goal_ip_permissions)
+	purge_rules        = module.params.get('purge_rules')
+	if goal_rules_size == 0 and not purge_rules:
```
Comment (heriet) [inline library/niftycloud_fw.py]: Why need `goal_rules_size == 0` ?

When add few ip_permissions and set `purge_rules = false`, user will expect to not revoke other rules.

see: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_group.py#L533

File: library/tests/test_niftycloud_fw.py [hunk 3]
```diff
@@ -1519,7 +1519,7 @@ def test_revoke_security_group_purge_skip(self):
 		)
 
 		mock_ip_permissions_removed = self.mockModule
-		mock_ip_permissions_removed['ip_permissions'] = []
+		mock_ip_permissions_removed.params['ip_permissions'] = []
```
Comment (heriet) [inline library/tests/test_niftycloud_fw.py]: It is better to `squash` this commit.

see: http://qiita.com/hiroponz/items/b7319f92581bd3facdc4

File: library/niftycloud_fw.py (hunk 2 above)
Comment (yuhei159) [inline library/niftycloud_fw.py]: `purge_rules` `s description is below.
> Purge existing rules on security group that are not found in rules

If there are no rules, this option remove rules.

File: library/tests/test_niftycloud_fw.py (hunk 3 above)
Comment (yuhei159) [inline library/tests/test_niftycloud_fw.py]: Thanks. I will use that as reference.

File: library/niftycloud_fw.py (hunk 1 above)
Comment (yuhei159) [inline library/niftycloud_fw.py]: I see.
This liburary use ip_permissions like ec2_group's `rules` and `rule_egress` , so I'll use `purge_ip_permissions`

File: library/niftycloud_fw.py (hunk 2 above)
Comment (yuhei159) [inline library/niftycloud_fw.py]: I see.
If there are extra previous rules on `purge_rules = false` , I'll don't revoke rules.

**Comment by heriet** [comment]
You should fix `ansible-role-niftycloud/library/documents/niftycloud_lb.md` about `purge_ip_permissions`.

**Comment by heriet** [comment]
LGTM!

**Pull request merged**